graspy/embed/oosase.py

```diff
@@ -0,0 +1,416 @@
+        initial embedding.
+    in_sample_idx : array-like , optional (default = None)
+    connected_attempts : integer , optional (default = 1000)
+        Number of sets of indices generated to find a connected induced subgraph.
+    semi_supervised : boolean , optional (default = False)
+        If true, once an out-of-sample vertex is embedded it is considered to be in-sample.
+    random_state : integer , optional (default = None)
+        Random seed used to generate in sample indices. If None, random_state
+        is a random integer between 0 and 10**6
+
+    Attributes
+    ----------
+    latent_left_ : array, shape (n_in_samples, n_components)
+        Estimated left latent positions of the graph.
+    singular_values_ : array, shape (n_components)
+        Singular values associated with the latent position matrices
+    in_sample_embedding : array, shape (n_in_samples, n_components)
+        Spectral embedding of the in sample vertices.
+    _nodes : array-like
+        List of node names
+    directed : bool
+        True if the graph is directed
+    id_to_index : dict
+        Mapping from node id to node index (i.e. 'node_id0' -> 0)
+    index_to_id : dict
+        Mapping from node index to node id (i.e. 0 -> 'node_id0')
+
+
+
+    See Also
+    --------
+    graspy.embed.selectSVD
+    graspy.embed.select_dimension
+
+    Notes
+    -----
+    The singular value decomposition: 
+
+    .. math:: A = U \Sigma V^T
+
+    is used to find an orthonormal basis for a matrix, which in our case is the adjacency
+    matrix of the graph. These basis vectors (in the matrices U or V) are ordered according 
+    to the amount of variance they explain in the original matrix. By selecting a subset of these
+    basis vectors (through our choice of dimensionality reduction) we can find a lower dimensional 
+    space in which to represent the graph.
+
+    References
+    ----------
+    .. [1] Sussman, D.L., Tang, M., Fishkind, D.E., Priebe, C.E.  "A
+       Consistent Adjacency Spectral Embedding for Stochastic Blockmodel Graphs,"
+       Journal of the American Statistical Association, Vol. 107(499), 2012
+    .. [2] Levin, K., Roosta-Khorasani, F., Mahoney, M. W., & Priebe, C. E. (2018). Out-of-sample 
+        extension of graph adjacency spectral embedding. PMLR: Proceedings of Machine Learning 
+        Research, 80, 2975-2984.
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        n_elbows=2,
+        algorithm="randomized",
+        n_iter=5,
+        check_lcc=True,
+        in_sample_proportion=1,
+        in_sample_idx=None,
+        connected_attempts=100,
+        semi_supervised=False,
+        random_state=None,
+    ):
+        super().__init__(
+            n_components=n_components,
+            n_elbows=n_elbows,
+            algorithm=algorithm,
+            n_iter=n_iter,
+            check_lcc=check_lcc,
+        )
+
+        if random_state is None:
```

Comment:
why set this? if `random_state` is None just use the system random state?

Reply:
sure